rpc/server.go

```diff
@@ -0,0 +1,53 @@
+package rpc
+
+import (
+	"fmt"
+	"net"
+	"net/rpc"
+
+	"github.com/gluster/glusterd2/context"
+	"github.com/gluster/glusterd2/peer"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/kshlm/pbrpc/pbcodec"
+)
+
+func (r *Connection) PeerAddRPCSvc(args *peer.PeerAddRequest, reply *RPCPeerAddResp) error {
+	log.Debug("In PeerAdd")
+	if context.MaxOpVersion < 40000 {
+		*reply.OpRet = -1
```

Comment:
`reply.OpRet` and `reply.OpError` are nil pointers, pointing to nothing. There is actually no location for the data you are setting to be stored. This is a side effect of using protobuf. Protobuf generated structs use only pointers as members. `net/rpc` package will allocate the struct and initialize the pointers to nil. The pointers will not be pointing to any allocated memory. You have to set these pointers to data you've allocated.

```go
var (
  opRet int32
  opError string
)

// Update opRet and opError instead of reply.OpRet and reply.OpError

// before returning
reply.OpRet = &opRet
reply.OpError = &opError

return nil
}